test/DGmethods/Euler/euler_model.jl

```diff
@@ -0,0 +1,146 @@
+using CLIMA.VariableTemplates
+using StaticArrays
+
+import CLIMA.DGmethods: BalanceLaw, vars_aux, vars_state, vars_gradient,
+                        vars_diffusive, flux!, source!, wavespeed,
+                        boundarycondition!, gradvariables!, diffusive!,
+                        init_aux!, init_state!, init_ode_param, init_ode_state
+
+abstract type EulerProblem end
+
+struct EulerModel{P, G, R} <: BalanceLaw
+  problem::P
+  gravity::G
+  refstate::R
+  function EulerModel(problem;
+                      gravity,
+                      refstate=NoReferenceState())
+    P = typeof(problem)
+    G = typeof(gravity)
+    R = typeof(refstate)
+    new{P, G, R}(problem,gravity, refstate)
+  end
+end
+
+init_state!(m::EulerModel, x...) = initial_condition!(m, m.problem, x...)
+
+function vars_aux(m::EulerModel, T)
+  NamedTuple{(:refstate, :gravity),
+             Tuple{vars_aux(m.refstate, T), vars_aux(m.gravity, T)}}
+end
+
+function vars_state(::EulerModel, T)
+  NamedTuple{(:ρ, :ρu⃗, :ρe), Tuple{T, SVector{3, T}, T}}
```

Comment:
In case you want to update this, Simon added a nice macro that allows you to write these functions as:
```
function vars_state(m::AtmosModel, T)
  @vars begin
    ρ::T
    ρu::SVector{3,T}
    ρe::T
    turbconv::vars_state(m.turbconv,T)
    turbulence::vars_state(m.turbulence,T)
    moisture::vars_state(m.moisture,T)
    radiation::vars_state(m.radiation,T)
  end
end
```